open-legends.lua

```diff
@@ -1,90 +1,123 @@
 -- open legends screen when in fortress mode
---@ module = true
 
 local dialogs = require('gui.dialogs')
 local gui = require('gui')
-local utils = require('utils')
+local widgets = require('gui.widgets')
 
-Restorer = defclass(Restorer, gui.Screen)
-Restorer.ATTRS{
-    focus_path='open-legends'
+-- --------------------------------
+-- LegendsManager
+--
+
+LegendsManager = defclass(LegendsManager, gui.Screen)
+LegendsManager.ATTRS {
+    focus_path='open-legends',
 }
 
-function Restorer:init()
-    print('initializing restorer')
-    self.region_details_backup = {} --as:df.world_region_details[]
-    local v = df.global.world.world_data.region_details
-    while (#v > 0) do
-        table.insert(self.region_details_backup, 1, v[0])
-        v:erase(0)
-    end
+function LegendsManager:init()
+    df.global.gametype = df.game_type.VIEW_LEGENDS
+
+    local legends = df.viewscreen_legendsst:new()
+    legends.page:insert("#",{new=true, header="Legends", mode=0, index=-1})
+    dfhack.screen.show(legends)
+
+    self:addviews{
+        widgets.Panel{
+            view_id='done_mask',
+            frame={t=1, r=1, w=9, h=3},
+        },
+    }
 end
 
-function Restorer:onIdle()
-    self:dismiss()
+function LegendsManager:render()
+    self:renderParent()
 end
 
-function Restorer:onDismiss()
-    print('dismissing restorer')
-    local v = df.global.world.world_data.region_details
-    while (#v > 0) do v:erase(0) end
-    for _,item in pairs(self.region_details_backup) do
-        v:insert(0, item)
+function LegendsManager:onInput(keys)
+    if keys.LEAVESCREEN or (keys._MOUSE_L and self.subviews.done_mask:getMousePos()) then
+        dialogs.showMessage('Exiting to avoid save corruption',
+            'Dwarf Fortress may be in a non-playable state\nand will now exit to protect your savegame.',
+            COLOR_YELLOW,
+            self:callback('dismiss'))
+        return true
     end
+    self:sendInputToParent(keys)
+    return true
 end
 
-function show_screen()
-    local ok, err = pcall(function()
-        Restorer{}:show()
-        dfhack.screen.show(df.viewscreen_legendsst:new())
-    end)
-    if not ok then
-        qerror('Failed to set up legends screen: ' .. tostring(err))
-    end
+function LegendsManager:onDismiss()
+    dfhack.run_command('die')
```

Comment:
I think players cannot be made sufficiently cognizant of the consequences of not restarting the game. I do not think we should offer a choice that will lead players to frustration later when they lose data. This tool is not niche. It will be used a lot by people with widely varying levels of understanding. I think it behooves us to guarantee safety to the best of our ability.

Reply:
The consensus (such as it is) on Discord is that giving players the option to (probably) corrupt their game is not a good move. I'll merge this as-is, but we can definitely change the behavior in the future.

Reply:
ack -- added https://github.com/DFHack/dfhack/issues/4424